libs/platform/kernel/ebpf_platform_kernel.c

```diff
@@ -761,6 +761,25 @@ ebpf_platform_thread_id()
     return (uint32_t)(uintptr_t)PsGetCurrentThreadId();
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL) _Must_inspect_result_ ebpf_result_t
+    ebpf_platform_get_authentication_id(_Out_ uint64_t* authentication_id)
+{
+    SECURITY_SUBJECT_CONTEXT context = {0};
+    SeCaptureSubjectContext(&context);
+    LUID local_authentication_id;
+
+    PACCESS_TOKEN access_token = SeQuerySubjectContextToken(&context);
+    ebpf_assert(access_token != NULL);
+
+    __analysis_assume(access_token != NULL);
+    NTSTATUS status = SeQueryAuthenticationIdToken(access_token, &local_authentication_id);
+    ebpf_assert(NT_SUCCESS(status));
+
+    *authentication_id = *(uint64_t*)&local_authentication_id;
+
+    return EBPF_SUCCESS;
```

Comment:
Was curious why we're ignoring the return values of the API calls and returning ```EBPF_SUCCESS``` un-conditionally (especially retail builds)?

Reply:
All of these APIs are expected to pass per MSDN. Hence this API is returning EBPF_SUCCESS. I will add a comment to this function clarifying why it is not checking the return status. For sanity, I have still added an ASSERT statement.

I can update the code if we feel we need to check the return values even if the APIs are expected to succeed.

Reply:
Added checks and also a comment stating these APIs are not expected to fail.